Foundation/Primitive/Types.hs

```diff
@@ -219,6 +221,36 @@ instance PrimType Word16 where
     {-# INLINE primAddrRead #-}
     primAddrWrite addr (Offset (I# n)) (W16# w) = primitive $ \s1 -> (# writeWord16OffAddr# addr n w s1, () #)
     {-# INLINE primAddrWrite #-}
+instance PrimType (BE Word16) where
```

Comment:
I think that `instance PrimType a => PrimType (BE a) where` (and same for LE) would have worked instead of explicitely adding each instances manually. `BE` and `LE` are just tag to represent that the value held inside is a different endianess. technically you could very well have a `BE Natural`, `BE Word128`, `BE CUInt` etc..